### ISSUE TITLE
UniqueKey.getReferences returns unnamed foreign keys

### ISSUE THREAD

**Issue by perceptron8** [jOOQ/jOOQ#7389]
### Problem description

After applying hot fix mentioned in #7377, I came across another problem:
`ForeignKey`s returned by `UniqueKey.getReferences` have `null` names.

The reason is that
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L880
invokes
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/ReferenceImpl.java#L76
instead of
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/ReferenceImpl.java#L83

### Proposed solution

It seems to me that

```java
references.add(new ReferenceImpl<Record, Record>(this, fkTable, fkFields));
```

could be simply replaced by something like

```java
String fkName = entry.getKey().get(3, String.class);
// ...
references.add(new ReferenceImpl<Record, Record>(this, fkTable, fkName, fkFields));
```

with `fkName` defined just after `fkTable`. 

### Final remarks

I believe that https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/ReferenceImpl.java#L76
should be removed entirely to eliminate similar bugs in the future.
It's not used anywhere else.

**Comment by lukaseder** [comment]
A belated thank you very much for your report. I agree with your assessments. We already have the `fkName` information in the result produced by JDBC's `DatabaseMetaData`, so we can just use it, and indeed, remove that particular constructor.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.12. Backport scheduled for 3.11.12 (#8734)
